library/nxos_command.py

```diff
@@ -0,0 +1,305 @@
+    body = data_dict['ins_api']['outputs']['output']['body']
+
+    return body
+
+
+def command_list_to_string(command_list):
+    """Converts list of commands into proper string for NX-API
+
+    Args:
+        cmds (list): ordered list of commands
+
+    Returns:
+        str: string of commands separated by " ; "
+
+    """
+    if command_list:
+        command = ' ; '.join(command_list)
+        return command + ' ;'
+    else:
+        return ''
+
+
+def send_command(device, cmds):
+    data_dict = xmltodict.parse(device.config(cmds)[1])
+    get_data = data_dict['ins_api']['outputs']['output']
+    try:
+        for each in get_data:
+            clierror = each.get('clierror', None)
+            msg = each.get('msg', None)
+            if clierror:
+                raise IOError(clierror, msg)
+    except AttributeError:
+            clierror = get_data.get('clierror', None)
+            msg = get_data.get('msg', None)
+
+    if clierror:
+        raise IOError(clierror, msg)
+    else:
+        body = get_data
+    return body
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            command=dict(required=False),
+            command_list=dict(required=False),
+            text=dict(choices=BOOLEANS, type='bool'),
+            type=dict(choices=['show', 'config'], required=True),
+            protocol=dict(choices=['http', 'https'], default='http'),
+            host=dict(required=True),
+            username=dict(type='str'),
+            password=dict(type='str')
+        ),
+        required_one_of=[['command', 'command_list']],
+        mutually_exclusive=[['command', 'command_list']],
+        supports_check_mode=False
+    )
+    if not HAS_PYCSCO:
+        module.fail_json(msg='There was a problem loading pycsco')
+
+    auth = Auth(vendor='cisco', model='nexus')
+    username = module.params['username'] or auth.username
+    password = module.params['password'] or auth.password
+    protocol = module.params['protocol']
+    host = socket.gethostbyname(module.params['host'])
+
+    command = module.params['command']
+    command_list = module.params['command_list']
+    text = module.params['text'] or None
+    cmd_type = module.params['type'].lower()
+
+    device = Device(ip=host, username=username, password=password,
+                    protocol=protocol)
+
+    if isinstance(command, list):
+        module.fail_json(msg='Only single command are supported using "command". \
+            If you want to use a List, go for "command_list" instead.')
+
+    if cmd_type == 'show':
+        if command_list:
+            module.fail_json(msg='Only single show commands are supported. \
+                Lists are only supported for config changes \
+                at this time.')
+
+    if cmd_type == 'config':
+        if isinstance(command_list, str):
+            module.fail_json(msg='Only Lists are supported when "command_list" \
+                is used.')
+        if text:
```

Comment:
I don't think we need the `if text:` -  if users supply it, it won't be used for anything anyway.